Directory.Build.targets

```diff
@@ -1,12 +1,30 @@
 <Project>
 
-<Target Name="SupportPrereleaseVersion"
-    AfterTargets="SetVersionFromChangelog">
-    <PropertyGroup Condition="'$(VersionSuffix)' != ''">
-        <Version>$(Version)-$(VersionSuffix)</Version>
-        <PackageVersion>$(Version)</PackageVersion>
-    </PropertyGroup>
-</Target>
+  <PropertyGroup>
+    <_BuildProjBaseIntermediateOutputPath>$(MSBuildThisFileDirectory)build/obj/</_BuildProjBaseIntermediateOutputPath>
```

Comment:
I need some Chetfu here to make this not hardcoded to build's immediate output path but i'm not sure exactly how to ask for a specific project's value.

Reply:
This is gonna be gross.  I think you'll have to have a 'GetIntermediateOutputPath` target in here that would return the value of the `BaseIntermediateOutputPath` property for the project it was part of. Then you'd call that target _on_ the build project during execution.

This dependency is why you often see features like this come in two targets:
* one to set up any data dependencies and generate the properties/paths for any inputs/outputs to the _real_ target
* and the the _real_ target the does the thing and uses the properties/items set up in the first target

this split is so that you always can set up your Inputs/Outputs and not run into issues with incrementality.

Reply:
@rainersigwald - we in the SDK get requests somewhat often for something like this: 'restore tools as part of my build'. Does anything here jump out as especially bad? Could we potentially extract some kind of recommendation in lieu of first class SDK level support?